### PR TITLE
[ARIFLOW-2458] Add cassandra-to-gcs operator

### PR DESCRIPTION
**Pull request by jgao54**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2458

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Create cassandra hook + cassandra_to_gcs operator that allows cql results to be stored in GCS in json format (compatible with BigQuery)

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests.contrib.operators.test_cassandra_to_gcs_operator
tests.contrib.operators.test_cassandra_hook
integration tests in local environment

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/operators/cassandra_to_gcs.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+from tempfile import NamedTemporaryFile
+from uuid import UUID
+
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.contrib.hooks.cassandra_hook import CassandraHook
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+PY3 = sys.version_info[0] == 3
```
Comment (Fokko) [inline airflow/contrib/operators/cassandra_to_gcs.py]: Can we use `six.py3` here?

**Review by Fokko — CHANGES_REQUESTED** [review]
@jgao54 Nice one. Small nit, can you also update the docs? Thanks!

**Comment by jgao54** [comment]
failure occured at db migration, unrelated to PR

**Review by Fokko — APPROVED** [review]
LGTM, thanks @jgao54

**Comment by Fokko** [comment]
Hmm, I don't really trust the failed tests, rebased against master: https://travis-ci.org/Fokko/incubator-airflow/builds/379609692

**Comment by jgao54** [comment]
Failure from PY3 envs  are on Snakebitem which is consistent with other recent PRs.
Failures from PY2 envs are on the following two tests that did not fail prior to rebase.
- test_scheduler_sla_miss_callback_exception
- test_scheduler_sla_miss_email_exception
Doesnt seem to suggest any potential correlation with this PR, kinda odd.

**Comment by jgao54** [comment]
Still failing one of

**Comment by bolkedebruin** [comment]
@jgao54 can you please limit the amount of closing and opening that you do? We have been contacted by Apache Infra explaining to us that we are taking up to 25% of total build time of Travis builds of all Apache projects. Please retry builds on your own account and point to successful builds there if necessary.

**Comment by jgao54** [comment]
@bolkedebruin I'll be more mindful, sorry about that.

**Pull request closed without merging**